packages/web-api-client/src/a11y-service-credential.spec.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// tslint:disable: no-null-keyword mocha-no-side-effect-code no-any no-unsafe-any
+import 'reflect-metadata';
+
+import { AuthenticationContext, TokenResponse } from 'adal-node';
+import * as request from 'request-promise';
+import { IMock, It, Mock, Times } from 'typemoq';
+
+import { A11yServiceCredential } from './a11y-service-credential';
+
+describe(A11yServiceCredential, () => {
+    let authenticationContextMock: IMock<AuthenticationContext>;
+    let testSubject: A11yServiceCredential;
+    let requestMock: IMock<typeof request>;
+    const clientId = 'client-id';
+    const clientSecret = 'client-sec';
+    const authorityUrl = 'authorityUrl';
+    const resource = 'resource-id';
+    const tokenResponse: TokenResponse = {
+        tokenType: 'type',
+        accessToken: 'at',
+    } as any;
+
+    let error: Error;
+
+    beforeEach(() => {
+        error = null;
+        requestMock = Mock.ofType<typeof request>(null);
+        authenticationContextMock = Mock.ofType<AuthenticationContext>();
+
+        testSubject = new A11yServiceCredential(clientId, clientSecret, resource, authorityUrl, authenticationContextMock.object);
+
+        authenticationContextMock
+            .setup(am => am.acquireTokenWithClientCredentials(resource, clientId, clientSecret, It.isAny()))
+            .returns((resourceUrl, cid, sec, callback) => {
+                callback(error, tokenResponse);
+            });
```

Comment:
move this to a helper method & then call it in each test method. (setupVerifiableGetTokenCall)

Reply:
Will quickly send a PR to address.